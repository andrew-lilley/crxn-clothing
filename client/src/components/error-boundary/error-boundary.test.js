import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './error-boundary.component';
import HomePage from '../../pages/homepage/homepage.component';

describe('ErrorBoundary component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ErrorBoundary><HomePage /></ErrorBoundary>);
  });

  test('should render ErrorBoundary component with no error', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render ErrorBoundary component with an error', () => {
    wrapper.setState({ hasErrored: true });
    expect(wrapper).toMatchSnapshot();
  });

});