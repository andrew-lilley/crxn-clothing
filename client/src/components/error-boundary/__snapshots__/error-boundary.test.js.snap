// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should render ErrorBoundary component with an error 1`] = `
<styled.div>
  <styled.div
    imageUrl="404.png"
  />
  <styled.h2>
    Our Dog Ate this Page
  </styled.h2>
  <styled.p>
    Our dog is cute but honestly a menace. Where are my trainers? Where is my jacket? Where is the hat that I was going to buy for my Auntâ€™s birthday? I hope that you can find what you are looking for in 
    <Link
      to="/shop"
    >
      our shop
    </Link>
  </styled.p>
</styled.div>
`;

exports[`ErrorBoundary component should render ErrorBoundary component with no error 1`] = `<HomePage />`;
